src/UserService.Business/Commands/Certificate/EditCertificateCommand.cs

```diff
@@ -23,111 +25,113 @@
-                        string.Join(',', response.Message.Errors));
-
-                    errors.Add(errorMessage);
-                }
-                else
-                {
-                    imageId = response.Message.Body;
-                }
-            }
-            catch (Exception exc)
-            {
-                _logger.LogError(exc, errorMessage);
-
-                errors.Add(errorMessage);
-            }
-
-            return imageId;
-        }
+          _logger.LogWarning("Can not add certificate image to certificate. Reason: '{Errors}'",
+            string.Join(',', response.Message.Errors));
 
-        public EditCertificateCommand(
-            IAccessValidator accessValidator,
-            IHttpContextAccessor httpContextAccessor,
-            IUserRepository userRepository,
-            ICertificateRepository certificateRepository,
-            IPatchDbUserCertificateMapper mapper,
-            IRequestClient<IAddImageRequest> rcImage,
-            ILogger<EditCertificateCommand> logger)
+          errors.Add(errorMessage);
+        }
+        else
         {
-            _accessValidator = accessValidator;
-            _httpContextAccessor = httpContextAccessor;
-            _userRepository = userRepository;
-            _certificateRepository = certificateRepository;
-            _mapper = mapper;
-            _rcImage = rcImage;
-            _logger = logger;
+          imageId = response.Message.Body;
         }
+      }
+      catch (Exception exc)
+      {
+        _logger.LogError(exc, errorMessage);
 
-        public OperationResultResponse<bool> Execute(Guid certificateId, JsonPatchDocument<EditCertificateRequest> request)
-        {
-            List<string> errors = new List<string>();
+        errors.Add(errorMessage);
+      }
+
+      return imageId;
+    }
 
-            var senderId = _httpContextAccessor.HttpContext.GetUserId();
-            var sender = _userRepository.Get(senderId);
+    public EditCertificateCommand(
+      IAccessValidator accessValidator,
+      IHttpContextAccessor httpContextAccessor,
+      IUserRepository userRepository,
+      ICertificateRepository certificateRepository,
+      IPatchDbUserCertificateMapper mapper,
+      ICreateImageDataMapper createImageDataMapper,
+      IRequestClient<ICreateImagesRequest> rcImage,
+      ILogger<EditCertificateCommand> logger)
+    {
+      _accessValidator = accessValidator;
+      _httpContextAccessor = httpContextAccessor;
+      _userRepository = userRepository;
+      _certificateRepository = certificateRepository;
+      _mapper = mapper;
+      _createImageDataMapper = createImageDataMapper;
+      _rcImage = rcImage;
+      _logger = logger;
+    }
 
-            DbUserCertificate certificate = _certificateRepository.Get(certificateId);
+    public OperationResultResponse<bool> Execute(Guid certificateId, JsonPatchDocument<EditCertificateRequest> request)
```

Comment:
make method async

Reply:
will be done in https://github.com/LT-Students/DigitalOffice-UserService/pull/122